util/terraform.go

```diff
@@ -128,98 +104,30 @@ func LoadVariablesFromSource(content, fileName, cwd string, applyTemplate bool,
-	for i, pattern := range patterns {
-		files, err := filepath.Glob(filepath.Join(folder, pattern))
-		if err != nil {
-			panic(err)
-		}
-		for _, file := range files {
-			matches[file] = i
-		}
-	}
-
-	// Then, we group files in two categories (regular and override) and we sort them alphabetically
-	var regularsFiles, overrideFiles []string
-	for file, index := range matches {
-		list := &regularsFiles
-		if index >= 2 {
-			// We group overrides files together
-			list = &overrideFiles
-		}
-		*list = append(*list, file)
-	}
-	sort.Strings(regularsFiles)
-	sort.Strings(overrideFiles)
-	return append(regularsFiles, overrideFiles...)
-}
-
-var patterns = []string{"*.tf", "*.tf.json", "override.tf", "override.tf.json", "*_override.tf", "*_override.tf.json"}
-
-func getDefaultVarsFromFile(filename string, unmarshal func([]byte, interface{}) error) (map[string]interface{}, error) {
-	bytes, err := ioutil.ReadFile(filename)
-	if err != nil {
-		return nil, err
-	}
-
-	content := make(map[string]interface{})
-	if err := unmarshal(bytes, &content); err != nil {
-		_, filename = filepath.Split(filename)
-		return nil, fmt.Errorf("%v %v", filename, err)
-	}
-
-	result, err := getDefaultVars(content)
-	if err != nil {
-		return result, fmt.Errorf("%v: %v", filename, err)
-	}
-
-	return result, nil
-}
-
-func getDefaultVars(content map[string]interface{}) (map[string]interface{}, error) {
-	var err error
-	result := make(map[string]interface{})
-
-	switch variables := content["variable"].(type) {
-	case map[string]interface{}:
-		addVariables(variables, result)
-	case []interface{}:
-		for _, value := range variables {
-			addVariables(value.(map[string]interface{}), result)
-		}
-	case nil:
-	default:
-		return nil, fmt.Errorf("Unknown variable type %[1]T: %[1]v", variables)
-	}
-
-	switch locals := content["locals"].(type) {
-	case map[string]interface{}:
-		result["local"] = locals
-	case []interface{}:
-		localMaps := make([]map[string]interface{}, len(locals))
-		for i := range localMaps {
-			localMaps[i] = locals[i].(map[string]interface{})
+func getTerraformVariableValues(terraformConfig interface{}, includeNil bool) (map[string]interface{}, error) {
+	variables := []*configs.Variable{}
+	switch source := terraformConfig.(type) {
+	case *configs.Module:
+		for _, variable := range source.Variables {
+			variables = append(variables, variable)
 		}
-		result["local"], err = utils.MergeDictionaries(localMaps...)
-	case nil:
-	default:
-		return nil, fmt.Errorf("Unknown local type %T", locals)
-	}
-
-	return result, err
-}
-
-func addVariables(source, target map[string]interface{}) {
-	for name, value := range source {
-		value := value.(map[string]interface{})
-		if value := value["default"]; value != nil {
-			target[name] = value
+	case *configs.File:
+		variables = source.Variables
+	}
+	variablesMap := map[string]interface{}{}
+	for _, variable := range variables {
+		if includeNil || !variable.Default.IsNull() {
+			var value interface{}
```

Comment:
can you add some inline comments on what is being done here?

Reply:
hmm. This is the code that I extracted to a cty helper file in the HCL2 PR. It is explained in that one